pylearn2/model_extensions/norm_constraint.py

```diff
@@ -21,12 +21,19 @@ class MaxL2FilterNorm(ModelExtension):
     Parameters
     ----------
     limit : float or symbolic float
-        The maximum column norm of the weight matrix is constrained
-        to be <= limit
+        The maximum norm of the weight matrix is constrained
+        to be <= limit along the axes
+    min_limit : float or symbolic float
+        The minimum norm of the weight matrix is constrained
+        to be => limit along the axes
+    axis : int or tuple of int
+        The axis or axes over which the norm is computed. Default is 0.
     """
 
-    def __init__(self, limit):
-        self.limit = limit
+    def __init__(self, limit, min_limit=0., axis=0):
+        self.max_limit = limit
+        self.min_limit = min_limit
+        self.axis = axis
```

Comment:
If both `min_limit` and `max_limit` are provided, can you assert that `min_limit <= max_limit`? The behaviour in the other case is ill-defined.